restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmConfigurationHandler.java

```diff
@@ -0,0 +1,326 @@
+            if (application == null) {
+                //create application
+                application = new FcmApplication("", this.applicationName);
+                application = fcmConfigurationClient.createApplication(application);
+            }
+
+            return  application;
+        }
+
+
+        private FcmCredentials getCredentials(String credentialsStorageString, FcmApplication application){
+            FcmCredentials credentials;
+
+            if(!TextUtils.isEmpty(credentialsStorageString)){
+                //get fcm credentials object
+                credentials = new FcmCredentials();
+                credentials.fillFromJson(credentialsStorageString);
+
+                if (credentials.getApplicationSid().equals(application.getSid())){
+                    return credentials;
+                }
+            }
+
+            credentials = fcmConfigurationClient.getCredentials(application);
+            //create new credentials
+            if (credentials == null){
+                credentials = new FcmCredentials("", application.getSid(), TYPE);
+                credentials = fcmConfigurationClient.createCredentials(credentials, fcmSecretKey);
+            }
+            return credentials;
+        }
+
+
+        @Override
+        protected void onPostExecute(Pair<HashMap<String, String>, RCClient.ErrorCodes> result) {
+            if(result.second == RCClient.ErrorCodes.SUCCESS){
+                HashMap<String, String> resultHash = result.first;
+                //save data to storage
+                String accountSid = resultHash.get(FCM_ACCOUNT_SID);
+                String clientSid = resultHash.get(FCM_CLIENT_SID);
+                String applicationString = resultHash.get(FCM_APPLICATION);
+                String credentialsString = resultHash.get(FCM_CREDENTIALS);
+                String bindingString = resultHash.get(FCM_BINDING);
+
+                RCLogger.v(TAG, "Storing RESULTS");
+                if (mStorageManagerWeak.get() != null) {
+                    mStorageManagerWeak.get().saveString(FCM_ACCOUNT_SID, accountSid);
+                    mStorageManagerWeak.get().saveString(FCM_CLIENT_SID, clientSid);
+                    mStorageManagerWeak.get().saveString(FCM_APPLICATION, applicationString);
+                    mStorageManagerWeak.get().saveString(FCM_CREDENTIALS, credentialsString);
+                    mStorageManagerWeak.get().saveString(FCM_BINDING, bindingString);
+                }
+
+
+                //if listener exists
+                if (mListenerWeak.get() != null) {
+                    mListenerWeak.get().onRegisteredForPush(RCClient.ErrorCodes.SUCCESS, RCClient.errorText(RCClient.ErrorCodes.SUCCESS));
+                }
+
+            } else {
+                RCClient.ErrorCodes errorCode = result.second;
+                if (mListenerWeak.get() != null) {
+                    switch (errorCode) {
```

Comment:
Do we need the switch and all these cases here? Can't we instead call onRegisteredForPush() for whichever error we have together with errorText for that error?

Reply:
of course not. I remove them. Thanks:)